### PR TITLE
E_CLASSROOM-303 [FE/BE] Category Hierarchy Page

### PR DESCRIPTION
**Pull request by Teraaay**
### Issue Link

https://framgiaph.backlog.com/view/E_CLASSROOM-303

### Definition of Done
* [x] Design is the same as the Figma design
* [x] Parent categories are displayed above the list
* [x] Clicking on one category will redirect to a page with all its subcategories
* [x] If subcategory text is too long, make it scrollable
* [x] (Low Prio) Make the breadcrumb headers clickable so that the user can go "back"

#### Main note, 
- The kebab menu is not included in this task.

- The Add/Edit Category Form is not in its original position [refer to below screenshot], so I had to make changes in its CSS file to fix that, please double check on your end as well.

> image

#### Pages Affected
- Category Hierarchy: http://localhost:3003/admin/category-hierarchy

### Screenshots
Category Hierarchy
> CATEGORY HIERARCHY

Fixed the position of the Add/Edit Category Form
> image

### CODE REVIEW

**Comment by catampongan** [comment]
## Approved PR #132 
- [x] Design is the same in Figma
- [x] When one category is clicked it redirects to its subcategories
- [x] Headers are clickable and redirect to its specific page
--------------
category hierarchy

--------------------
- Total Test Time: 0.25 hr
- Total Iterations: 0
- Total Comments: 0

File: src/pages/admin/categories/CategoryHierarchy/index.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      ? toast('Processing', 'Getting the subcategories...')
+      : toast('Processing', 'Getting the root categories...');
+
+    CategoryApi.getCategories({ category_id: chosenCategoryPathID })
+      .then(({ data }) => {
+        setCategories(data.data);
+      })
+      .catch(() =>
+        toast('Error', 'There was an error getting the list of categories.')
+      );
```
Comment (karlolee50) [inline src/pages/admin/categories/CategoryHierarchy/index.js]: you can put this in a load() function

File: src/pages/admin/categories/CategoryHierarchy/index.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        ...breadcrumbs,
+        { id: category.id, name: category.name, idx: index }
+      ]);
+    } else {
+      toast('Message', 'This category does not have a subcategory.');
+    }
+  };
+
+  const onBreadcrumbClick = (category_id, index) => {
+    console.log(breadcrumbs);
```
Comment (karlolee50) [inline src/pages/admin/categories/CategoryHierarchy/index.js]: remove if not used

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 10 minutes
Comment/s: 2

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**